### PR TITLE
[JUJU-2289] Rewrite remove-application unit tests

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Rewrite remove-application unit tests to use GoMock generated mocks

Also, as a flyby, make some common sense simplifications

A couple of JujuConnSuite tests were dropped, because they were testing things outside of this package

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

```sh
go test github.com/juju/juju/cmd/juju/application
```

### CODE REVIEW

File: cmd/juju/application/package_test.go [hunk 1]
```diff
@@ -12,6 +12,9 @@ import (
 
 // TODO(wallyworld) - convert tests moved across from commands package to not require mongo
```
Comment (manadart) [inline cmd/juju/application/package_test.go:13]: Is this considered done based in this patch?

**Review by manadart — APPROVED** [review]

File: cmd/juju/application/package_test.go [hunk 2]
```diff
@@ -12,6 +12,9 @@ import (
...
 
+//go:generate go run github.com/golang/mock/mockgen -package mocks -destination mocks/applicationapi_mock.go github.com/juju/juju/cmd/juju/application ApplicationAPI,RemoveApplicationAPI
+//go:generate go run github.com/golang/mock/mockgen -package mocks -destination mocks/deployer_mock.go github.com/juju/juju/cmd/juju/application/deployer Deployer,DeployerFactory
+
 func TestPackage(t *stdtesting.T) {
 	testing.MgoTestPackage(t)
```
[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply cmd/juju/application/package_test.go:19]: Unfortunately there are still some JujuConnSuite tests here, so it is still required

File: cmd/juju/application/package_test.go (hunk 1 above)
Comment (jack-w-shaw) [reply cmd/juju/application/package_test.go:13]: Not yet unfortunately

File: cmd/juju/application/removeapplication.go [hunk 3]
```diff
@@ -197,11 +188,9 @@ func (c *removeApplicationCommand) removeApplications(
...
 ) error {
 	var maxWait *time.Duration
-	if c.Force {
-		if c.NoWait {
-			zeroSec := 0 * time.Second
-			maxWait = &zeroSec
-		}
+	if c.NoWait {
+		zeroSec := 0 * time.Second
+		maxWait = &zeroSec
```
[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply cmd/juju/application/removeapplication.go:193]: Worth discussing, potentially at the standup go-around. But feels a bit out of scope for this PR, so I'm going to go ahead and merge

**Pull request merged**